onnxruntime/core/providers/webnn/builders/helper.cc

```diff
@@ -99,69 +99,77 @@ bool IsTensorShapeSupported(const NodeArg& node_arg, const std::string& parent_n
   return true;
 }
 
-// Check if all input tensor ranks of the given node are supported by WebNN.
-bool IsInputRankSupportedByOp(const Node& node, const emscripten::val& wnn_limits, const logging::Logger& logger) {
-  const std::string_view op_type = node.OpType();
-  const auto it = op_inputs_map.find(op_type);
-  if (it == op_inputs_map.end()) {
-    LOGS(logger, VERBOSE) << "Operator type: [" << op_type << "] is not found in the op inputs map.";
+// Helper to check if a single input's rank is supported
+bool IsInputRankSupported(const emscripten::val& wnn_limits,
+                          const std::string_view webnn_op_type,
+                          const std::string_view input_name,
+                          const int input_rank,
+                          const std::string& node_name,
+                          const logging::Logger& logger) {
+  const std::string webnn_op_type_str(webnn_op_type);
+  const std::string input_name_str(input_name);
+
+  if (wnn_limits[webnn_op_type_str].isUndefined() ||
+      wnn_limits[webnn_op_type_str][input_name_str].isUndefined()) {
+    LOGS(logger, VERBOSE) << "Node name: [" << node_name
+                          << "], WebNN op type: [" << webnn_op_type
+                          << "], input [" << input_name
+                          << "] limits are not defined in wnn_limits.";
     return false;
   }
 
-  const auto& input_defs = node.InputDefs();
-  const std::string_view webnn_op_type = it->second.opType;
-  const std::string webnn_op_type_str(webnn_op_type);
+  const emscripten::val& input_limits = wnn_limits[webnn_op_type_str][input_name_str];
```

Comment:
```suggestion
  const emscripten::val input_limits = wnn_limits[webnn_op_type_str][input_name_str];
```
It's not a pointer, you don't need the `&`. BTW, you can move this definition before line 112.

Reply:
Why remove this line before the ` wnn_limits[webnn_op_type_str][input_name_str].isUndefined()` check, should it be defined after checking the validation of ` wnn_limits[webnn_op_type_str][input_name_str].isUndefined()`